pyaedt/edb.py

```diff
@@ -3356,3 +3358,145 @@ def calculate_initial_extent(self, expansion_factor):
+                os.mkdir(working_directory)
+            else:
+                shutil.rmtree(working_directory)
+                os.mkdir(working_directory)
+        else:
+            working_directory = os.path.dirname(self.edbpath)
+        zone_primitives = list(self.active_layout.GetZonePrimitives())
+        edb_zones = {}
+        if not self.setups:
+            self.siwave.add_siwave_syz_analysis()
+            self.save_edb()
+        for zone in zone_primitives:
+            edb_zone_path = os.path.join(
+                working_directory, "{}_{}".format(zone.GetId(), os.path.basename(self.edbpath))
+            )
+            shutil.copytree(self.edbpath, edb_zone_path)
+            poly_data = zone.GetPolygonData()
+            edb_zones[edb_zone_path] = poly_data
+        return edb_zones
+
+    @pyaedt_function_handler()
+    def cutout_multizone_layout(self, zone_dict, common_reference_net):
+        """Create multizone project cutout.
+
+        Parameters
+        ----------
+        zone_dict : dict[str](EDB PolygonData)
+            Dictionary with EDB path as key and EDB PolygonData as value defining the zone region.
+            This dictionary is returned from the command copy_zones():
+            >>> edb = Edb(edb_file)
+            >>> zone_dict = edb.copy_zones(r"C:\Temp\test")
+
+        common_reference_net : str
+            the common reference net name. This net name must be provided to provide a valid project.
+
+        Returns
+        -------
+        dict[str][str] , dict[str][str]
+        first dictionary defined_ports with edb name as key and existing port name list as value
+        second dictionary terminals with edb name as key and created ports name on clipped signal nets
+
+        """
+        terminals = {}
+        defined_ports = {}
+        for edb_path, zone_polygon in zone_dict.items():
+            edb = Edb(edbversion=self.edbversion, edbpath=edb_path)
+            edb.stackup.stackup_mode = "Laminate"
+            signal_nets = list(self.nets.signal.keys())
+            edb.cutout(use_pyaedt_cutout=True, custom_extent=zone_polygon, open_cutout_at_end=True)
+            edb.active_cell.SetName(os.path.splitext(os.path.basename(edb_path))[0])
+            defined_ports[os.path.splitext(os.path.basename(edb_path))[0]] = list(edb.excitations.keys())
+            edb_terminals_info = edb.hfss.create_vertical_circuit_port_on_clipped_traces(
+                nets=signal_nets, reference_net=common_reference_net, user_defined_extent=zone_polygon
+            )
+            if edb_terminals_info:
+                terminals[os.path.splitext(os.path.basename(edb_path))[0]] = edb_terminals_info
+            edb.save_edb()
+            edb.close_edb()
+        return defined_ports, terminals
+
+    @pyaedt_function_handler()
+    def get_connected_ports_from_multizone_cutout(self, terminal_info_dict):
+        """Return connected port list from clipped multizone layout.
```

Comment:
```suggestion
        """Get the list of connected ports from the clipped multizone layout.
```

Reply:
Again, I'm not sure why you didn't accept my change. If the method uses "Get", it's best to use "Get in the description". In general, you want to avoid long strings of adjectives--no more than 2, which is why I suggested the change to "Get the list of ....." In this case, you have only two adjectives (connected and port), so we could say Get the connected port list...."